crates/core/src/kernel/models/actions.rs

```diff
@@ -134,10 +148,72 @@ pub fn contains_timestampntz<'a>(mut fields: impl Iterator<Item = &'a StructFiel
     fields.any(|f| _check_type(f.data_type()))
 }
 
+pub(crate) trait ProtocolExt {
```

Comment:
Please please please add some rustdoc here. This is a clever way to shove more functionality on the kernel `Protocol`, but important to document